### PR TITLE
JIRA 18844 - adds error handling for the details page in replication

### PR DESCRIPTION
**Pull request by seanfeld**
# Fix 18844 - handle errors on replication details page in replication

## Description
actually handle errors instead of just console log them

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [18844](https://cdap.atlassian.net/browse/CDAP-18844)

### CODE REVIEW

File: app/cdap/components/Replicator/Detail/index.tsx [hunk 1]
```diff
@@ -104,6 +105,7 @@ interface IDetailState {
   setTimeRange: (timeRange: string) => void;
   exportPipeline: () => void;
   duplicatePipeline: () => void;
+  error: object | string | null;
```
Comment (mrahanjam) [inline app/cdap/components/Replicator/Detail/index.tsx:108]: Not sure about the error object props, but shouldn't we send `error.response` or `error.message` (whichever is in the error object) and remove `object` type here (i.e., restrict it to string | null) for better readability.

Comment (seanfeld) [reply app/cdap/components/Replicator/Detail/index.tsx:108]: error banner handles that

Comment (seanfeld) [reply app/cdap/components/Replicator/Detail/index.tsx:108]:
```
  const errorMessage: string =
    objectQuery(error, 'response', 'message') || objectQuery(error, 'response') || error;
    ```

Comment (mrahanjam) [reply app/cdap/components/Replicator/Detail/index.tsx:108]: Ah I see 👍

**Review by mrahanjam — APPROVED** [review]
LGTM!

File: app/cdap/components/Replicator/Detail/index.tsx (hunk 1 above)
Comment (seanfeld) [reply app/cdap/components/Replicator/Detail/index.tsx:108]: If we wanted to do this right, the API handler should handlers should all return the same type of error and we shouldn't have to worry about it

**Pull request merged**